### ISSUE TITLE
How to convert tensorflow model to .wts

### ISSUE THREAD

**Issue by pribadihcr** [zhaozhixu/SqueezeDetTrt#1]
Hi @zhaozhixu,

How to convert tensorflow checkpoint model to sqdtrt.wts

Thanks

**Comment by zhaozhixu** [comment]
Hi @pribadihcr , 

You can use these two scripts:
[](https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts_tf.py)
[](https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts.pl)
Put them in the same folder and run `python genwts_tf.py -h` to see the usage. You have to keep the tensor names consistent with weight names in [sqdtrt.cpp](https://github.com/zhaozhixu/SqueezeDetTRT/blob/master/sqdtrt.cpp). (All scope separators '/' are replaced to '_'.)

Actually [LightNet](https://github.com/zhaozhixu/LightNet) would be a better solution for general neural network models.

**Comment by zhaozhixu** [comment]
Hi @pribadihcr ,

You can use these two scripts:
[https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts_tf.py](https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts_tf.py)
[https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts.pl](https://github.com/zhaozhixu/LightNet/blob/master/tools/genwts.pl)

Put them in the same folder and run `python genwts_tf.py -h` to see the usage. You have to keep the tensor names consistent with weight names in [sqdtrt.cpp](https://github.com/zhaozhixu/SqueezeDetTRT/blob/master/sqdtrt.cpp) (All scope separators '/' are replaced to '_').

Actually [LightNet](https://github.com/zhaozhixu/LightNet) would be a better solution for general neural network models.

**Comment by pribadihcr** [comment]
Hi @zhaozhixu ,

Thanks. for LightNet does it support squeezedet also.

**Comment by zhaozhixu** [comment]
@pribadihcr Yes. Its design goal is to support the optimization and deployment on various backends for general models. Some demoes with TensorRT including squeezedet have been finished and will be committed soon.

**Comment by pribadihcr** [comment]
Hi @zhaozhixu,

Why my converted model .wts has bigger size?.. its 36.9 MB. The checkpoint TF model is aorund 17 MB. But the converted model seems work. Thanks

**Comment by zhaozhixu** [comment]
@pribadihcr Because TensorRT uses literal hexadecimal strings to store weights. That means two chars for one byte, so it's about twice the size.
